check-compatibility-with-govuk-frontend-version.js

```diff
@@ -0,0 +1,37 @@
+if (process.env.HMRC_FRONTEND_DISABLE_COMPATIBILITY_CHECK === 'true') {
+  process.exit(0);
+}
+const fs = require('fs');
+const path = require('path');
+
+// To our knowledge hmrc-frontend won't work with versions below this.
+// This is because of a change to directory structure (/dist/).
+const minimumGovukFrontendVersion = '5.1.0';
+
+const projectDir = process.env.INIT_CWD;
+const projectPackagePath = path.join(projectDir, 'package.json');
+
+const projectPackageJson = JSON.parse(fs.readFileSync(projectPackagePath, 'utf8'));
+const currentVersionInProject = projectPackageJson.dependencies['govuk-frontend'].replace(/[^0-9.]/g, '');
+if (currentVersionInProject >= minimumGovukFrontendVersion) {
```

Comment:
I don't think we're comparing the semver strings correctly here, string comparison will look correct some of the time, but it's just comparing digit by digit - not part by part, so for example:

```js
("9.2.0" > "9.1.0") == true
("10.0.0" > "9.0.0") == false
("10.99.0" > "10.19.0") == true
("10.100.0" > "10.19.0") == false
```

would be good to have some test cases like these, and then I think we'd need to split the semver string and compare parts one at a time

Reply:
here's an example (in scala) of how to compare that I had to do recently https://github.com/hmrc/platui/pull/10/files#diff-e803cb6aad6d0e19a39f9224e7491f0f249dcc9ee92844cfd0cf37d2b18b5dfaR61-R69

some logic like this to see if X semver is greater than Y semver
split the parts of the semver strings, for each part, if X.major == Y.major then compare minor, otherwise return X.major > Y.major

Reply:
we also could just use https://www.npmjs.com/package/semver - since it's from npm - don't feel too bad about adding a new dependency on it for this 🤷 maybe - though when I say it out loud I'm like "noo"

Reply:
and apparently because of the fact that 0 == false in js, you can write it very succinctly!

```js
function semverGreaterThan(a, b) {
  const [aMaj, aMin, aPat] = a.split('.').map(Number);
  const [bMaj, bMin, bPat] = b.split('.').map(Number);
  return ((aMaj - bMaj) || (aMin - bMin) || (aPat - bPat)) > 0;
};
```